src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -140,7 +140,9 @@ public String getDescriptionAndLink() {
      * org.jabref.gui.externalfiletype.ExternalFileTypes#getExternalFileTypeByName(String)}
      */
     public JabRefIcon getTypeIcon() {
-        return IconTheme.JabRefIcons.PDF_FILE;
+        return externalFileTypes.fromLinkedFile(linkedFile, false)
```

Comment:
Can you please also remove the above TODO

Reply:
Ok, I get it. I will finish it in one hour. Many thank for your tips